### ISSUE TITLE
hyperparameter tuning for KNN

### ISSUE THREAD

**Issue by Goutham-R** [Goutham-R/special-umbrella#2]

**Comment by NarayanAdithya** [comment]
knn=KNeighborsClassifier(n_neighbors=1)
knn.fit(X_train,y_train)
y_pred=knn.predict(X_test)
print(mean_squared_error(y_test,y_pred)) 
print(classification_report(y_test,y_pred))

#n_neighbors=2 0.57
knn=KNeighborsClassifier(n_neighbors=2)
knn.fit(X_train,y_train)
y_pred=knn.predict(X_test)
print(mean_squared_error(y_test,y_pred)) 
print(classification_report(y_test,y_pred))

#n_neighbors=3 0.57
knn=KNeighborsClassifier(n_neighbors=3)
knn.fit(X_train,y_train)
y_pred=knn.predict(X_test)
print(mean_squared_error(y_test,y_pred)) 
print(classification_report(y_test,y_pred))

#n_neighbors=4 0.58
knn=KNeighborsClassifier(n_neighbors=4)
knn.fit(X_train,y_train)
y_pred=knn.predict(X_test)
print(mean_squared_error(y_test,y_pred)) 
print(classification_report(y_test,y_pred))

#n_neighbors=5 0.56
knn=KNeighborsClassifier(n_neighbors=5)
knn.fit(X_train,y_train)
y_pred=knn.predict(X_test)
print(mean_squared_error(y_test,y_pred)) 
print(classification_report(y_test,y_pred))

import  matplotlib.pyplot  as plt
arr=list(range(1,60))
score=[]
from sklearn.model_selection import cross_val_score
for i in arr:
    knn=KNeighborsClassifier(n_neighbors=i)
    scores=cross_val_score(knn,X_train,y_train,cv=10,scoring='accuracy')
    score.append(scores.mean())
max(score)
